contracts/bepro/Network_Registry.sol

```diff
@@ -7,36 +7,57 @@ import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "../utils/ReentrancyGuardOptimized.sol";
 import "./INetwork_v2.sol";
 import "../utils/Governed.sol";
+import "../math/SafePercentMath.sol";
 
 contract Network_Registry is ReentrancyGuardOptimized, Governed {
+
     constructor(IERC20 _erc20,
-                uint256 _lockAmountForNetworkCreation,
-                address _treasury,
-                uint256 _lockFeePercentage) ReentrancyGuardOptimized() Governed() {
+        uint256 _lockAmountForNetworkCreation,
+        address _treasury,
+        uint256 _lockFeePercentage,
+        uint256 _closeFee,
+        uint256 _cancelFee) ReentrancyGuardOptimized() Governed() {
         erc20 = IERC20(_erc20);
         lockAmountForNetworkCreation = _lockAmountForNetworkCreation;
         treasury = _treasury;
         lockFeePercentage = _lockFeePercentage;
+        closeFee = _closeFee;
+        cancelFee = _cancelFee;
     }
 
     using SafeMath for uint256;
 
     INetwork_v2[] public networksArray;
     IERC20 public erc20;
 
+
+    struct AllowedToken {
+        address _address;
+        uint256 id;
+    }
+
+    mapping(address => address) public allowedTransactionalTokens;
```

Comment:
Why is a not an array ? and have a settter for transactional and reward tokens.

Reply:
@hvasconcelos so we _don't have to loop_ on an array when we create a bounty as looping would raise the gas considerably (along with the length of the array)

This way we have a 1-to-1 mapping _and_ have a way to _retrieve_ the information by using the array below (`allowedTokens`)